virttest/migration_template.py

```diff
@@ -365,16 +431,25 @@ def _post_start_vm(self):
 
     def _migrate(self):
         """
-        1.Set selinux state
-        2.Record vm uptime
-        3.For postcopy migration:
+        0.Create disk image on dest host
+        1.Set --xml and --persistent-xml in migrate options
+        2.Set selinux state
+        3.Record vm uptime
+        4.For postcopy migration:
             1) Set migration speed to low value
             2) Monitor postcopy event
-        4.Do live migration
-        5.Check migration result: succeed or fail with expected error
-        6.For postcopy migration: check postcopy event
-        7.Do post migration check: check vm state, uptime, network
+        5.Do live migration
+        6.Check migration result: succeed or fail with expected error
+        7.For postcopy migration: check postcopy event
+        8.Do post migration check: check vm state, uptime, network
         """
+        # Create disk image on dest host if --copy-storage-all/inc is used
```

Comment:
Here why we need to create some disk images on target host?
In my previous test, only create a dir on target host is enough and the source image will be copied to target dir, isn't it?

Reply:
1. Yes, if we don't create an image on dest host manually, libvirt will create it automatically. But there was a bug in this case: [Bug 1248996 ](https://bugzilla.redhat.com/show_bug.cgi?id=1248996)- Larger size of qcow2 image file on target host after migration with non-shared storage with full disk copy. The consequence of the bug is that full image is copied to dest host even if only part of the disk is allocated, which waste a lot of time during test.

2. In layered product, it pre-creates image on dest host by itself, not by libvirt.

Based on above two reasons, I add "create image" step in automation script.

Reply:
thx for the explanation, LGTM